src/Generation/GapicClientGenerator.php

```diff
@@ -392,6 +402,45 @@ private function construct(): PhpClassMember
         $options = AST::var('options');
         $optionsParam = AST::param($this->ctx->type(Type::array()), $options, AST::array([]));
         $clientOptions = AST::var('clientOptions');
+
+        // Assumes there are only two transport types.
+        $isGrpcRest = $this->serviceDetails->transportType === Transport::GRPC_REST;
+
+        $restTransportDocText = 'At the moment, supports only `rest`.';
+        $grpcTransportDocText = 'May be either the string `rest` or `grpc`. Defaults to `grpc` if gRPC support is detected on the system.';
+        $transportDocText =
+          PhpDoc::text(
+              'The transport used for executing network requests. ',
+              $isGrpcRest ? $grpcTransportDocText : $restTransportDocText,
+              '*Advanced usage*: Additionally, it is possible to pass in an already instantiated',
+            // TODO(vNext): Don't use a fully-qualified type here.
+            $ctx->type(Type::fromName(TransportInterface::class), true),
+              'object. Note that when this object is provided, any settings in $transportConfig, and any $serviceAddress',
+              'setting, will be ignored.'
+          );
+
+        $transportConfigSampleValues = [];
+        if ($isGrpcRest) {
+            $transportConfigSampleValues['grpc'] = AST::arrayEllipsis();
+        }
+        // Set this value here, don't initialize it, so we can maintain alphabetical order
+        // for the resulting printed doc.
+        $transportConfigSampleValues['rest'] = AST::arrayEllipsis();
+        $transportConfigDocText =
+          PhpDoc::text(
+              'Configuration options that will be used to construct the transport. Options for',
+              'each supported transport type should be passed in a key for that transport. For example:',
+              PhpDoc::example(AST::block(
+                  AST::assign(AST::var('transportConfig'), AST::array($transportConfigSampleValues))
+              ), null, true),
+              'See the',
+              AST::call($ctx->type(Type::fromName($isGrpcRest ? GrpcTransport::class : RestTransport::class), true), AST::method('build'))(),
```

Comment:
nit: This is hard to follow... is it possible to split this out?

Reply:
Done.